olm/catalogd/v1alpha1/clustercatalog_types.go

```diff
@@ -0,0 +1,223 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SourceType defines the type of source used for catalogs.
+// +enum
+type SourceType string
+
+const (
+	SourceTypeImage SourceType = "Image"
+
+	TypeProgressing = "Progressing"
+	TypeServing     = "Serving"
+
+	// Serving reasons
+	ReasonAvailable   = "Available"
+	ReasonUnavailable = "Unavailable"
+	ReasonDisabled    = "Disabled"
+
+	// Progressing reasons
+	ReasonSucceeded = "Succeeded"
+	ReasonRetrying  = "Retrying"
+	ReasonBlocked   = "Blocked"
+
+	MetadataNameLabel = "olm.operatorframework.io/metadata.name"
+
+	AvailabilityEnabled  = "Enabled"
```

Comment:
Enabled/Disabled needs more dimension, and communicate better to those with less op-con context. 
Proposed were something like "AvailableToBeUsed" and its reverse.